### ISSUE TITLE
Optimize BlueChi's systemd units on boot

### ISSUE THREAD

**Issue by engelmi** [engelmi/hirte#864]
## Please describe what you would like to see

Currently, the `bluechi-agent` and `bluechi-controller` are started `After=network.target`. This [special target](https://www.freedesktop.org/software/systemd/man/latest/systemd.special.html#network.target) is only weakly defined and there is 
> no guarantee that hardware-based devices have shown up by the time this target is reached, or even acquired complete IP configuration

As an alternative, [network-online.target](https://www.freedesktop.org/software/systemd/man/latest/systemd.special.html#network-online.target) is mentioned. However, it states there: 
> What precisely this requires is left to the implementation of the network managing service.

As well as:

> Note that networking daemons that simply provide functionality to other hosts (as opposed to consume functionality of other hosts) generally do not need to pull this in.

Since both BlueChi components, controller and agent, require only the DBus to be up and running locally, the ordering on boot can be optimized to `After=dbus.service`. 

## Please describe the solution you'd like

The systemd units of `bluechi-agent` and `bluechi-controller` are changed to: 

```ini
[Unit]
# ...
DefaultDependencies=no
Before=basic.target shutdown.target
After=dbus.service
Conflicts=shutdown.target
# ...
```

**Comment by alexlarsson** [comment]
I don't think you even need dbus.service. All it needs dbus.socket, but i believe that is a dependency of basic.target via sockets.target, and basic.target is a default dependency for all services.

**Comment by alexlarsson** [comment]
Ah, I see you drop the default deps, but still you only need dbus.socket to be able to start bluechi. It will then block startup until dbus-broker accepts the connection.

**Comment by alexlarsson** [comment]
Honestly though, I'm not sure starting before basic.target is right. I wouldn't recommend disabling the default dependencies, as you will basically have a half-configured system at that point.

**Comment by engelmi** [comment]
I agree and think we shouldn't implement this. If such optimizations are needed, these optimizations can be done outside of BlueChi itself. Lets close this.
